src/reactjs/components/DomainEditor/DomainEditor.tsx

```diff
@@ -2,165 +2,164 @@
-  showMessage?: boolean;
-  message?: string;
-  type?: string;
-  isEdit?: boolean;
+  openFlyout?: boolean
+  updateDomainGrid?: boolean
+  selectedDomain?: any
+  showPopup?: boolean
+  showMessage?: boolean
+  message?: string
+  type?: string
+  isEdit?: boolean
 }
 
 export class DomainEditor extends React.Component<domainProps, domainState> {
-  constructor(props: domainProps) {
-    super(props);
+  constructor (props: domainProps) {
+    super(props)
     this.state = {
       openFlyout: false,
       updateDomainGrid: false,
-      selectedDomain: "",
+      selectedDomain: '',
       showPopup: false,
       showMessage: false,
-      message: "",
-      type: "",
+      message: '',
+      type: '',
       isEdit: false
-    };
+    }
   }
 
-  getSelectedDomain = (domain) =>
+  getSelectedDomain = (domain): void =>
     this.setState({
-      selectedDomain: domain,
-    });
+      selectedDomain: domain
+    })
 
-  togglePopup = () => this.setState({ showPopup: !this.state.showPopup });
+  togglePopup = (): void => this.setState({ showPopup: !isFalsy(this.state.showPopup) })
 
-  handleChange = () => this.setState({ openFlyout: !this.state.openFlyout, isEdit: false });
+  handleChange = (): void => this.setState({ openFlyout: !isFalsy(this.state.openFlyout), isEdit: false })
 
-  handleEdit = () => this.setState({openFlyout: !this.state.openFlyout, isEdit: true})
+  handleEdit = (): void => this.setState({ openFlyout: !isFalsy(this.state.openFlyout), isEdit: true })
 
-  confirmDelete = async () => {
-    const { rpsKey } = this.context.data;
-    let domainName = encodeSpecialCharacters(this.state.selectedDomain[0].name);
-    const response = await HttpClient.delete(`${this.props.rpsServer}/api/v1/admin/domains/${domainName}`, rpsKey);
+  confirmDelete = async (): Promise<any> => {
+    const { rpsKey } = this.context.data
+    const domainName = encodeSpecialCharacters(this.state.selectedDomain[0].name)
+    const response = await HttpClient.delete(`${String(this.props.rpsServer)}/api/v1/admin/domains/${domainName}`, rpsKey)
 
     if (response === `Domain ${domainName} successfully deleted`) {
       this.setState({
         showMessage: true,
         message: response,
-        type: `success`,
-        updateDomainGrid: !this.state.updateDomainGrid,
-        selectedDomain: "",
-        showPopup: !this.state.showPopup,
-      });
+        type: 'success',
+        updateDomainGrid: !isFalsy(this.state.updateDomainGrid),
+        selectedDomain: '',
+        showPopup: !isFalsy(this.state.showPopup)
+      })
     } else {
       this.setState({
         showMessage: true,
         message: response,
-        showPopup: !this.state.showPopup,
-        type: `error`,
-      });
+        showPopup: !isFalsy(this.state.showPopup),
+        type: 'error'
+      })
     }
-    this.showNotification();
-  };
+    this.showNotification()
+  }
 
-  showNotification = () =>
+  showNotification = (): any =>
     setTimeout(() => {
```

Comment:
this should be void?

Reply:
Since setTimeout function returns a value, the return type has been updated to match the return type of setTimeout function.